src/app/containers/MostRead/utilities/mostReadShape.js

```diff
@@ -1,20 +1,41 @@
-import { string, shape, arrayOf, number } from 'prop-types';
+import { string, shape, arrayOf, number, oneOfType } from 'prop-types';
+
+const cpsMostReadRecord = {
+  id: string.isRequired,
+  promo: shape({
+    timestamp: number.isRequired,
+    headlines: shape({
+      shortHeadline: string.isRequired,
+    }),
+    locators: shape({
+      assetUri: string.isRequired,
+    }),
+  }),
+};
+
+const optimoMostReadRecord = {
+  id: string.isRequired,
+  timestamp: number.isRequired,
+  locators: shape({
+    canonicalUrl: string.isRequired,
+  }),
+  headlines: shape({
+    promoHeadline: shape({
+      blocks: arrayOf(
+        shape({
```

Comment:
This nested structure makes me a little nervous, is there a schema we can check? That said, this nested block structure is how we get back Optimo data elsewhere, it just seems like a lot to do this on the front end to access this string. If this is the data we're getting then we can't really change this though!

Reply:
A schema exists on confluence, however it might be out of date.